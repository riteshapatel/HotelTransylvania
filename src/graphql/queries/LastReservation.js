import gql from 'graphql-tag';

const LastReservation = gql`
query lastReservation { 
    lastReservationMadeAt, lastReserved @client
}
`
export default LastReservation;