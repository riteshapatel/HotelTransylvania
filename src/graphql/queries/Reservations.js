/**
 * reservations query
 * @author ritesh.patel
 */
import gql from 'graphql-tag';

const Reservations = gql`
query Reservations (
        $where: ReservationWhereInput!
    ){
        reservations(
            where: $where
        ) {
            id
            name 
            hotelName 
            arrivalDate 
            departureDate 
        }
    
    }
`
export default Reservations;