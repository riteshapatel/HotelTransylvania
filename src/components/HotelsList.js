/**
 * @class HotelsList component 
 * @author ritesh.patel
 * App landing screen
 */
import React, { Component } from 'react';
import { Text, View, Image, FlatList } from 'react-native';
import CreepyButton from './CreepyButton';
import Reservations from '../graphql/queries/Reservations.js';
import LastReservation from '../graphql/queries/LastReservation';
import CreepyItem from './CreepyItem';
import { Query, graphql, ApolloConsumer } from 'react-apollo';
import styles from '../styles/Global.style.js';
import gql from 'graphql-tag';

class HotelsList extends Component {
    render () { 
        const {navigate} = this.props.navigation;
        const name = 'Ritesh';

        return (
            <View>
                <Query
                    query={Reservations} 
                    variables={{where: {name: 'Ritesh'}}}
                    pollInterval={500}
                >
                    {({ loading, error, data }) => {
                        if (loading) return (<Text>Loading...</Text>)
                        if (error) return (<Text>Error...{error}</Text>)

                        return (
                            <FlatList data={data.reservations}
                                renderItem={({item}) => 
                                    <View>
                                        <CreepyItem hotelName={item.hotelName} navigate={navigate} data={item}/>
                                    </View>
                                } 
                                keyExtractor={(item, index) => index.toString()} 
                            />
                        )
                    }}
                </Query>
                <Query
                    query={LastReservation}>
                    {({ data }) => {
                        let displayText = '';
                        if (data.lastReservationMadeAt === 'None') {
                            displayText = '** No reservations made yet **'
                        } else {
                            displayText = `${data.lastReserved} @ ${data.lastReservationMadeAt}`;
                        }
                        return (
                            <Text style={styles.last_reservation_text}>{displayText}</Text>
                        )
                    }}
                </Query>
            </View>
        )
    }
}
export default HotelsList;