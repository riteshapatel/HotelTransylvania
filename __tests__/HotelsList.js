/**
 * Unit test for HotelsList component 
 * @author ritesh.patel
 */
import React from 'react';
import {MockedProvider} from 'react-apollo/test-utils';
import Reservations from '../src/graphql/queries/Reservations';
import LastReservation from '../src/graphql/queries/LastReservation';
import renderer from 'react-test-renderer';
import HotelsList from '../src/components/HotelsList';

test ('snapshot HotelsList component', () => {
    const navigation = { navigate: jest.fn() }
    const mocks = [
        {
            request: {
                query: Reservations,
                variables: {
                    where: {
                        name: "Ritesh"
                    }
                }
            },
            result: {
                data: {
                    reservations: [{
                        name: '',
                        hotelName: '',
                        arrivalDate: '',
                        departureDate: '',
                        id: '',
                        __typename: ''
                    }]
                }
            }
        },
        {
            request: {
                query: LastReservation
            },
            result: {
                data: {
                    lastReservationMadeAt: '',
                    lastReserved: ''
                }
            }
        }
    ]
    const tree = renderer.create(<MockedProvider mocks={mocks}><HotelsList navigation={navigation} /></MockedProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});``