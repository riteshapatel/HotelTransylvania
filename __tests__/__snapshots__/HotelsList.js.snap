// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot HotelsList component 1`] = `
<View>
  <Text>
    Loading...
  </Text>
  <Text
    style={
      Object {
        "fontSize": 12,
        "fontStyle": "italic",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    undefined @ undefined
  </Text>
</View>
`;
